Ignore event spams of keys we already have

neithernut opened the pull request:
This PR fixes a bug where keeping a key down for a while causes the DAG not to be traverses correctly.

Commit:
Ignore event spams of keys we already have

A key press should only cause a state change if the key wasn't pressed
before.

matthiasbeyer commented:
Will test, stay tuned!

matthiasbeyer commented on src/input/hotkeys.c:
```diff
@@ -118,9 +118,27 @@ ws_hotkeys_eval(
+                break;
+            }
+            // we consume the event but we'll _not_ store it or iterate the DAG
+            return empty_eventlist();
```
Wouldn't it result in less code if you'd do a `if (buf->code == code) { return ... }` here and remove the `continue` therefor?

neithernut replied:
The extra level of intendation would make the documentation in line 137 a pain in the ass.

matthiasbeyer replied:
Meh. Could be fixed using a multi-line arg, but well.. I won't argue about that.

matthiasbeyer commented:
ACK.

matthiasbeyer commented:
**STOP!**

Not ACK. Press a key twice crashes the system.

neithernut commented:
May this be related to #514? (Will incluse a fix in this PR btw.)

Commit:
Add missing range-check in hotkey_dag_get_next()

neithernut commented:
Please check whether there are still problems.

TheNeikos commented:
I can't reproduce it

TheNeikos commented:
#515 is still open, do you think that is related?

Commit:
Check whether interested in release event before processing it

This saves us a little bit of overhead.
It also prevents waysome from producing a deceptive log messge.

neithernut commented:
Issue #515 _should_ now be resolved (I only tested it briefly).

matthiasbeyer commented:
ACK.
